test/performance/connection/export_performance_test.py

```diff
@@ -0,0 +1,122 @@
+import csv
+from pathlib import Path
+from typing import (
+    Callable,
+    Optional,
+)
+
+import pandas as pd
+import polars as pl
+import pytest
+from pyarrow import dataset
+
+from pyexasol import ExaConnection
+
+
+def _get_csv_length(filepath: Path) -> int:
+    with Path(filepath).open(mode="r", encoding="utf-8") as file:
+        reader = csv.reader(file)
+        return len(list(reader))
+
+
+def _get_parquet_length(filepath: Path) -> int:
+    parquet_dataset = dataset.dataset(filepath)
+    return sum(
+        row_group.num_rows
+        for fragment in parquet_dataset.get_fragments()
+        for row_group in fragment.row_groups
+    )
+
+
+def _get_list_length(exported_list: list) -> int:
+    return len(exported_list)
+
+
+def _get_pandas_dataframe_length(exported_pandas: pd.DataFrame) -> int:
+    return exported_pandas.shape[0]
+
+
+def _get_polars_dataframe_length(exported_polars: pl.DataFrame) -> int:
+    return exported_polars.shape[0]
+
+
+@pytest.mark.parametrize(
+    "export_method, file_length_function",
+    [
+        pytest.param("export_to_list", _get_list_length, id="list"),
+        pytest.param("export_to_pandas", _get_pandas_dataframe_length, id="pandas"),
+        pytest.param("export_to_polars", _get_polars_dataframe_length, id="polars"),
+    ],
+)
+def test_export_methods_to_memory(
+    benchmark,
+    benchmark_specs,
+    connection: ExaConnection,
+    columns,
+    fill_sales_table,
+    export_method: str,
+    file_length_function: Callable,
+):
+    def func_to_be_measured():
+        return getattr(connection, export_method)(
+            query_or_table=fill_sales_table,
+            query_params={"columns": columns},
+        )
+
+    result = benchmark.pedantic(
+        func_to_be_measured, iterations=1, rounds=benchmark_specs.rounds
+    )
+
+    number_rows = file_length_function(result)
+    assert number_rows == benchmark_specs.final_data_size
+
+
+@pytest.mark.parametrize(
+    "export_method, file_extension, file_length_function, callback_params",
+    [
+        pytest.param("export_to_file", "csv", _get_csv_length, None, id="csv"),
```

Comment:
I almost tend to use a data class as parameters, this shortly before confusing many parameters

Reply:
If it's alright, for now, I'd leave it like this. When it's added to, will keep this in mind as a nicer pattern.